### PR TITLE
Fixing veracode reader condition check

### PR DESCRIPTION
**Pull request by 0xgoto**
I was trying to check the score for the scan result that I got using Veracode. Benchmark has the support for this, but still, it was not generating a scorecard. The next format of Veracode results has an empty line (Second line). But the Veracode Reader takes the empty line for checking whether it is a Veracode result or not.

In the older format, the data starts from line 2, whereas now the data starts from line 3. So I have added a condition that supports both the old and new formats.

### CODE REVIEW

**Comment by darkspirit510** [comment]
Thank you for the PR. If possible, can you reduce the test file so it's easier to understand?

And, if you don't mind/are allowed to, can you send me the old and new result file to github@darkspirit510.de, i'd also like to add a test for the old format and compare them.

**Comment by 0xgoto** [comment]
I have reduced the test file and also mailed you the example files for the new and old formats

**Comment by darkspirit510** [comment]
Thank you, the files work for me. Can you go through my other comments, too?

**Review by darkspirit510 — CHANGES_REQUESTED** [review]

File: plugin/src/test/java/org/owasp/benchmarkutils/score/parsers/VeracodeReaderTest.java
Comment (0xgoto) [reply plugin/src/test/java/org/owasp/benchmarkutils/score/parsers/VeracodeReaderTest.java:22]: There is an issue with the ReaderTestBase file. When I try to change the condition check logic to checking the root node name, the ReaderTestBase is always throwing "FileNotFound" exception. I have tested the reader by building it, was able to generate the scorecard. So, I am removing the test for this Reader alone.

Comment (darkspirit510) [reply plugin/src/test/java/org/owasp/benchmarkutils/score/parsers/VeracodeReaderTest.java:22]: Yes, this is because of the way the reader accesses the data. In VeracodeReader, please change

InputSource is = new InputSource(new FileInputStream(resultFile.file()));

to

InputSource is = new InputSource(new StringReader(resultFile.content()));

Comment (0xgoto) [reply plugin/src/test/java/org/owasp/benchmarkutils/score/parsers/VeracodeReaderTest.java:22]: Figured it and solved. will push the code

**Comment by darkspirit510** [comment]
Just checked out and testet your changes. It works well, thank you!

One little thing: Can you please run `mvn compile` and commit your testfile afterwards? The maven task applies some linting to the test file.

Now it's up to @davewichers 😉

**Review by darkspirit510 — APPROVED** [review]

**Comment by 0xgoto** [comment]
I ran `mvn compile`. It ran successfully and there is no changes to commit

**Comment by darkspirit510** [comment]
That's weird...

```
sknoop:BenchmarkUtils/ (pr-veracode) $ gss
sknoop:BenchmarkUtils/ (pr-veracode) $ mvn compile
[...]
[INFO] BUILD SUCCESS
[...]
sknoop:BenchmarkUtils/ (pr-veracode) $ gss
 M plugin/src/test/resources/testfiles/Benchmark_Veracode.xml
```

**Comment by 0xgoto** [comment]
What to do?
```
❯ mvn compile
[...]
[INFO] BUILD SUCCESS
[...] 
❯ git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean

```

**Comment by darkspirit510** [comment]
I don't know, why it remains unchanged for you. Just ignore it, I can fix it later. The rest is fine, we just have to wait for Dave's approval. 😊

**Pull request merged**
